src/Symfony/Component/VarDumper/Caster/SocketCaster.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @internal
+ */
+final class SocketCaster
+{
+    public static function castSocket(\Socket $socket, array $a, Stub $stub, bool $isNested): array
+    {
+        socket_getsockname($socket, $addr, $port);
+        $info = stream_get_meta_data(socket_export_stream($socket));
+
+        $a += [
+            Caster::PREFIX_VIRTUAL.'address' => $addr,
+            Caster::PREFIX_VIRTUAL.'port' => $port,
+            Caster::PREFIX_VIRTUAL.'eof' => $info['eof'],
+            Caster::PREFIX_VIRTUAL.'mode' => $info['mode'],
+            Caster::PREFIX_VIRTUAL.'seekable' => $info['seekable'],
```

Comment:
are those really populated? looking at php-src I could find where

Reply:
Yes, `SocketCasterTest` asserts on them. I also won't be able to tell where in php-src however...

Reply:
I meant populated by php-src. To me, those info are going to be always the same for all sockets, so this could be noise.
The entry point in the source is:
https://github.com/php/php-src/blob/2473f57ba7c0ceeb6c08dfe335995925d0497028/ext/sockets/sockets.c#L2386

Reply:
Alright so we should instead just keep `blocked` and `timeout`

Reply:
yes, and maybe for uri, add addr+port ?
`tcp://0.0.0.0:0` (or `tcp://[::]:0` if AF_INET6)
$a[Caster::PREFIX_VIRTUAL.'uri'] = $info['uri'] ?? '//').\sprintf(str_contains($addr, ':') ? '[%s]:%s' : '%s:%s', $addr, $port);

Reply:
Yes! [The test](https://github.com/symfony/symfony/pull/59035/files#diff-1d15eccf7a7d450c4ce07e0fb153e58c8909994d394eb1da90bbb7622070d013) now does reflect this change. Thanks for the hint!